public/tier0/threadtools.h

```diff
@@ -595,22 +591,22 @@ class CInterlockedPtr
 class PLATFORM_CLASS CThreadMutex
 {
 public:
-	CThreadMutex();
+	CThreadMutex( const char* pDebugName );
 	~CThreadMutex();
 
 	//------------------------------------------------------
 	// Mutex acquisition/release. Const intentionally defeated.
 	//------------------------------------------------------
-	void Lock();
-	void Lock() const		{ (const_cast<CThreadMutex *>(this))->Lock(); }
-	void Unlock();
-	void Unlock() const		{ (const_cast<CThreadMutex *>(this))->Unlock(); }
+	void Lock( const char *pFileName, int nLine );
+	void Lock( const char *pFileName, int nLine ) const		{ (const_cast<CThreadMutex *>(this))->Lock( pFileName, nLine ); }
+	void Unlock( const char *pFileName, int nLine );
+	void Unlock( const char *pFileName, int nLine ) const	{ (const_cast<CThreadMutex *>(this))->Unlock( pFileName, nLine ); }
 
-	bool TryLock();
-	bool TryLock() const	{ return (const_cast<CThreadMutex *>(this))->TryLock(); }
+	bool TryLock( const char *pFileName, int nLine );
+	bool TryLock( const char *pFileName, int nLine ) const	{ return (const_cast<CThreadMutex *>(this))->TryLock( pFileName, nLine ); }
 
-	void LockSilent(); // A Lock() operation which never spews.  Required by the logging system to prevent badness.
-	void UnlockSilent(); // An Unlock() operation which never spews.  Required by the logging system to prevent badness.
+	void LockSilent( const char *pFileName, int nLine ); // A Lock() operation which never spews.  Required by the logging system to prevent badness.
+	void UnlockSilent( const char *pFileName, int nLine ); // An Unlock() operation which never spews.  Required by the logging system to prevent badness.

```

Comment:
I see that you've added the params in here, but I don't see them being used anywhere down the line? Can you explain why that change is required and how did you come up with it?

Reply:
These parameters may have been used in the debug build, or at least they are in the exported names. So I added them according to this prototype.

![cs2-thread](https://github.com/alliedmodders/hl2sdk/assets/63844820/fc1e5968-3f38-4634-a132-5f6c579849dc)
![cs2-thread2](https://github.com/alliedmodders/hl2sdk/assets/63844820/7f33b062-0f4e-46e7-81ef-0faae1bdca3c)

Reply:
Which binary is that?

Reply:
Okay yeah, see them in the tier0.dll